src/ProjectService.Business/Commands/CreateTaskCommand.cs

```diff
@@ -0,0 +1,118 @@
+using MassTransit;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Logging;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace LT.DigitalOffice.ProjectService.Business.Commands
+{
+    public class CreateTaskCommand : ICreateTaskCommand
+    {
+        private readonly ITaskRepository _repository;
+        private readonly ICreateTaskValidator _validator;
+        private readonly IDbTaskMapper _mapperTask;
+        private readonly IHttpContextAccessor _httpContextAccessor;
+        private readonly IRequestClient<IGetDepartmentRequest> _requestClient;
+        private readonly IAccessValidator _accessValidator;
+        private readonly IProjectRepository _projectRepository;
+        private readonly ILogger<CreateTaskCommand> _logger;
+
+        private IGetDepartmentResponse GetDepartment(Guid userId, List<string> errors)
+        {
+            string errorMessage = "Cannot create task. Please try again later.";
+
+            try
+            {
+                var response = _requestClient.GetResponse<IOperationResult<IGetDepartmentResponse>>(
+                    IGetDepartmentRequest.CreateObj(userId, null)).Result;
+
+                if (response.Message.IsSuccess)
+                {
+                    return response.Message.Body;
+                }
+
+                _logger.LogWarning($"Can not find department with this id '{userId}':", userId);
+            }
+            catch (Exception exc)
+            {
+                _logger.LogError(exc, errorMessage);
+
+                errors.Add(errorMessage);
+            }
+
+            return null;
+        }
+
+        public CreateTaskCommand(
+            ITaskRepository repository,
+            ICreateTaskValidator validator,
+            IProjectRepository projectRepository,
+            IDbTaskMapper mapperTask,
+            IHttpContextAccessor httpContextAccessor,
+            IAccessValidator accessValidator,
+            IRequestClient<IGetDepartmentRequest> requestClient,
+            ILogger<CreateTaskCommand> logger)
+        {
+            _repository = repository;
+            _validator = validator;
+            _mapperTask = mapperTask;
+            _httpContextAccessor = httpContextAccessor;
+            _requestClient = requestClient;
+            _accessValidator = accessValidator;
+            _logger = logger;
+            _projectRepository = projectRepository;
+        }
+
+        public OperationResultResponse<Guid> Execute(CreateTaskRequest request)
+        {
+            var errors = new List<string>();
+
+            var authorId = _httpContextAccessor.HttpContext.GetUserId();
+
+            DbProject project = _projectRepository.GetProject(request.ProjectId);
+
+            IGetDepartmentResponse department = GetDepartment(authorId, errors);
+
+            bool isAdmin = _accessValidator.IsAdmin();
+
+            bool isProjectParticipant = project?.Users.FirstOrDefault(x =>
```

Comment:
do you do select in getDepartment method? If you don't do it, projects.Users will be always null